File: app/src/main/java/com/vishal/redditclone/utils/RedditCloneApplication.java
package com.vishal.redditclone.utils;

import android.app.Application;

public class RedditCloneApplication extends Application {
    private static RedditCloneApplication instance;

    public static RedditCloneApplication getAppInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        instance = this;
        super.onCreate();
    }
}

File: app/src/main/java/com/vishal/redditclone/utils/RedditCloneApplication.kt
package com.vishal.redditclone.utils

import android.app.Application

class RedditCloneApplication : Application() {
	override fun onCreate() {
		appInstance = this
		super.onCreate()
	}

	companion object {
		var appInstance: RedditCloneApplication? = null
			private set
	}
}